src/shortest-path/test/astar.js

```diff
@@ -0,0 +1,89 @@
+/**
+ * Graphology A* Shortest Path Unit Tests
+ * =============================================
+ */
+var assert = require('assert'),
+  library = require('../astar.js'),
+  graphology = require('graphology');
+
+var UndirectedGraph = graphology.UndirectedGraph,
+  DirectedGraph = graphology.DirectedGraph;
+
+var EDGES = [
+  ['1', '2', 1],
+  ['1', '8', 3],
+  ['2', '3', 2],
+  ['2', '4', 3],
+  ['4', '5', 2],
+  ['5', '6', 1],
+  ['6', '7', 5],
+  ['7', '8', 4],
+  ['7', '9', 1],
+  ['8', '9', 10],
+  ['10', '11', 3]
+];
+
+var heuristic = (a, b) => 0
+
+describe('astar', function () {
```

Comment:
Did you test the case where a proper heuristic is given?

Reply:
No. I wasn't really sure how to do that. I guess the way to test that would be to check which paths were explored. But I don't know how to do that.

Alternatively, I could give this a heuristic as a "smoke test", and check that it still finds the shortest path, but not actually test whether it made more efficient.

Reply:
I am not sure I remember well but isn't a A* pathfinding without relevant heuristic just the same as Dijkstra? In which case, don't you use a proper heuristic when you use it? Else why would it be faster than Dijkstra?

Reply:
I understand why it could be difficult to test that A* is "working" without peeking in its insides indeed. However, a paranoid test with a nontrivial heuristic function could be nice, such as the one networkx uses: https://github.com/networkx/networkx/blob/main/networkx/algorithms/shortest_paths/tests/test_astar.py

Maybe it is also possible to design a test where there exists two equivalent path that dijkstra would choose arbitrarily but that the A* heuristic forces to choose, but it might be a hassle and I would not refuse the PR in the absence of such a test in any case.

My confusion came from this statement of yours:

> While working on a project, I needed shortest-path to work faster.

I guess my question was more about your particular use-case where you actually needed A* to compute shortest paths faster and I was befuddled when your answer was:

> No. I wasn't really sure how to do that. I guess the way to test that would be to check which paths were explored. But I don't know how to do that.

because I did not understand it was related to the "testability" of the A* mechanism, as you demonstrated further afterwards. Sorry for my misunderstanding here.

Reply:
Ah. Yes, I "tested" it in my own code (outside this library). In my use-case, I am using a graph of streets from OpenStreetMap, and I want to find the shortest path of >100k point pairs. I used a heuristic of euclidean distance, and it was WAY faster to use A* than Dijkstra. I didn't measure before/after, but it was something like several minutes before, and a few seconds after.